zebra/zebra_nb_config.c

```diff
@@ -1323,32 +1326,59 @@ int lib_interface_zebra_bandwidth_destroy(enum nb_event event,
 }
 
 /*
- * XPath: /frr-vrf:lib/vrf/frr-zebra:ribs/rib
+ * XPath: /frr-vrf:lib/vrf/frr-zebra:zebra/ribs/rib
  */
-int lib_vrf_ribs_rib_create(enum nb_event event, const struct lyd_node *dnode,
-			    union nb_resource *resource)
+int lib_vrf_zebra_ribs_rib_create(enum nb_event event,
+				  const struct lyd_node *dnode,
+				  union nb_resource *resource)
 {
+	struct vrf *vrf;
+	afi_t afi = AFI_IP;
+	safi_t safi = SAFI_UNICAST;
+	struct zebra_vrf *zvrf;
+	struct zebra_router_table *zrt;
+
+	vrf = nb_running_get_entry(dnode, NULL, false);
+	zvrf = vrf_info_lookup(vrf->vrf_id);
+
+	zrt = zebra_router_find_zrt(zvrf, zvrf->table_id, afi, safi);
+
 	switch (event) {
 	case NB_EV_VALIDATE:
+		if (!zrt) {
+			if (IS_ZEBRA_DEBUG_EVENT)
+				zlog_debug("%s: vrf %s table is not found.",
+					   __PRETTY_FUNCTION__, vrf->name);
+			return NB_ERR_VALIDATION;
+		}
+		break;
 	case NB_EV_PREPARE:
 	case NB_EV_ABORT:
+		break;
 	case NB_EV_APPLY:
-		/* TODO: implement me. */
+
+		nb_running_set_entry(dnode, zrt);
+
 		break;
 	}
 
 	return NB_OK;
 }
 
-int lib_vrf_ribs_rib_destroy(enum nb_event event, const struct lyd_node *dnode)
+int lib_vrf_zebra_ribs_rib_destroy(enum nb_event event,
+				   const struct lyd_node *dnode)
 {
-	switch (event) {
-	case NB_EV_VALIDATE:
-	case NB_EV_PREPARE:
-	case NB_EV_ABORT:
-	case NB_EV_APPLY:
-		/* TODO: implement me. */
-		break;
+	if (event != NB_EV_APPLY)
+		return NB_OK;
+
+	struct zebra_router_table *zrt;
+
+	zrt = nb_running_unset_entry(dnode);
+	if (!zrt) {
```

Comment:
this is unnecessary, `nb_running_unset_entry()` will abort if it fails (which should never happen unless the .create()/.destroy() callbacks aren't in sync).